Controller/Process/Result.php

```diff
@@ -390,11 +390,30 @@ protected function validatePayloadAndReturnResponse($response)
         $request = [];
 
         if (!empty($this->_session->getLastRealOrder()) &&
-            !empty($this->_session->getLastRealOrder()->getPayment()) &&
-            !empty($this->_session->getLastRealOrder()->getPayment()->getAdditionalInformation("paymentData"))
+            !empty($this->_session->getLastRealOrder()->getPayment())
         ) {
-            $request['paymentData'] = $this->_session->getLastRealOrder()->getPayment()->
-            getAdditionalInformation("paymentData");
+
+            if (!empty($this->_session->getLastRealOrder()->getPayment()->getAdditionalInformation('paymentData'))) {
+                $request['paymentData'] = $this->_session->getLastRealOrder()->getPayment()->
+                getAdditionalInformation('paymentData');
+
+                // remove paymentData from db
+                $this->_session->getLastRealOrder()->getPayment()->unsAdditionalInformation('paymentData');
+                $this->_session->getLastRealOrder()->getPayment()->save();
+            }
+
+            // for pending payment that redirect we store this under adyenPaymentData
+            // we need to refactor the code in the plugin that all paymentData is stored in paymentData and not in adyenPaymentData
```

Comment:
maybe marking this as a todo or with a story number so we remember to remove this comment when this is done?
```suggestion
            // TODO: refactor the code in the plugin that all paymentData is stored in paymentData and not in adyenPaymentData
```